wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/service/ExecuteCommandService.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.languageserver.service;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import org.eclipse.che.api.core.jsonrpc.commons.JsonRpcException;
+import org.eclipse.che.api.languageserver.exception.LanguageServerException;
+import org.eclipse.che.api.languageserver.registry.InitializedLanguageServer;
+import org.eclipse.che.api.languageserver.registry.LanguageServerRegistry;
+import org.eclipse.che.api.languageserver.shared.model.ExtendedExecuteCommandParams;
+import org.eclipse.che.api.languageserver.util.LSOperation;
+import org.eclipse.che.api.languageserver.util.OperationUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Json RPC API for the textDoc to send workspace/executeCommand requests. */
+@Singleton
+public class ExecuteCommandService {
```

Comment:
I'm not sure it's a good idea to expose a general command executor service. The commands we're interested in are specific extensions for jdt.ls.
Also, if we did expose this functionality, WorkspaceService would be a more natural place for it.

Reply:
It's for now we need execute commands just for jdt.ls, maybe this functionality will be useful for other language servers. About `WorkspaceService` it is more like handler of events which were sent from che-client, that's because I moved executeCommand to different service.

Reply:
Nope, part of workspace service: https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#workspace_executeCommand